driver/Src/CAN.c

```diff
@@ -0,0 +1,377 @@
+#include "stm32xx_hal.h"
+#include "CAN.h"
+
+// 8 for now unless extended payload is supported
+#define DATA_SIZE (8)
+
+// entries in queues
+typedef struct {
+    CAN_TxHeaderTypeDef header;
```

Comment:
Shouldn't this and the RXHeaderTypeDef be pointers so that the user can define the memory once and it doesn't have to be copied in every payload

Reply:
Never mind, figured this out. Confused it with the HandleTypeDef